packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenLockRolesWithManager.sol

```diff
@@ -0,0 +1,337 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import "../../../access-control-registry/RoleDeriver.sol";
+import "../../../access-control-registry/AccessControlClient.sol";
+import "../../../access-control-registry/interfaces/IAccessControlRegistry.sol";
+import "./interfaces/IAirnodeTokenLockRolesWithManager.sol";
+
+/// @title Contract that implements generic AccessControlRegistry roles for
+/// the AirnodeTokenLock contract
+contract AirnodeTokenLockRolesWithManager is
+    RoleDeriver,
+    AccessControlClient,
+    IAirnodeTokenLockRolesWithManager
+{
+    // There are Six roles implemented in this contract:
+    // Root
+    // └── (1) Admin (can grant and revoke the roles below)
+    //     ├── (2) Oracle address setter
+    //     ├── (3) coefficient and registry setter
```

Comment:
I can imagine a case where we can expect an address to need to be able to update the coefficient but not the fee registry addresss, so I think this should be broken down for consistency